common/djangoapps/third_party_auth/saml.py

```diff
@@ -219,8 +219,12 @@ def get_attr(self, attributes, conf_key, default_attribute):
         another attribute to use.
         """
         key = self.conf.get(conf_key, default_attribute)
-        default = self.conf['attr_defaults'].get(conf_key) or None
-        return attributes[key][0] if key in attributes else default
+        if key in attributes:
+            try:
+                return attributes[key][0]
+            except IndexError:
+                log.error(u'SAML attribute "%s" value not found.', key)
```

Comment:
Given that this is somewhat expected, can we drop this log level down to info or warning?